app/views/layouts/_notifications.html.erb

```diff
@@ -38,7 +38,12 @@
         </div>
       <% end %>
       <div class="d-flex align-items-center justify-content-center mb-2 mt-3">
+        <% if @unread.count.positive? %>
+          <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text-left" %>
```

Comment:
This will redirect the user to the `notification` page, but if the user clicks on `mark all as read` from a different page it should stick to that page only!

You need to write another method, route for this one @VaibhavUpreti

Reply:
Added another route and defined another method ... now after reading notifications from the modal user sticks to the same page.